### ISSUE TITLE
"'const' is not defined"

### ISSUE THREAD

**Issue by garyking** [typescript-eslint/typescript-eslint#2560]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JS
module.exports = {
  extends: [],
  parser: '@typescript-eslint/parser',
  plugins: ['@typescript-eslint'],
  rules: {}
}
```

```TS
const item = {
  a: 'a'
} as const;

export default item;
```

**Expected Result**

No error.

**Actual Result**

`'const' is not defined  no-undef`

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `4.1.1` |
| `TypeScript`                | `4.0.2` |
| `ESLint`                    | `7.9.0` |
| `node`                      | `14.8.0` |

**Comment by bradzacher** [comment]
We have passing tests for this in master:

https://github.com/typescript-eslint/typescript-eslint/blob/f785965a12e20e82de1c7f3d039023c225115eb1/packages/eslint-plugin/tests/eslint-rules/no-undef.test.ts#L223-L226

As this was fixed in #2499.

This implies that likely you have multiple versions of our tooling installed at the same time.

**Comment by garyking** [comment]
`rm -rf node_modules/` then `yarn install` did the trick.
